drivers/timer/Kconfig.ch32v00x

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2024 Michael Hope
+# SPDX-License-Identifier: Apache-2.0
+
+config CH32V00X_SYSTICK
+	bool "CH32V00X systick timer"
+	depends on SOC_CH32V003
+	default y
+	depends on DT_HAS_WCH_SYSTICK_ENABLED
```

Comment:
This is a good example.
Is this only valid for the SOC_CH32V003 or for the whole dts bind DT_HAS_WCH_SYSTICK_ENABLED?
I assume you want to drop the SOC_CH32V003 entry.